### ISSUE TITLE
`no-multiple-resolved` wrongly reports `reject` in try/catch block

### ISSUE THREAD

**Issue by FloEdelmann** [eslint-community/eslint-plugin-promise#382]
### Description

The new `no-multiple-resolved` rule (#369) reports an error if `resolve` is used in the `try` block and `reject` is used in the `catch` block, even though they can't be both executed:

```js
return new Promise((resolve, reject) => {
  https.get(someUrl, response => {
    response.on(`end`, async () => {
      try {
        await writeFile(someFilePath, someContents);
        resolve();
      }
      catch (error) {
        reject(error);
      }
    });
  });
});
```

### Steps to Reproduce

(see code above)

**Expected behavior:** No error, since `resolve` is only called when the `writeFile` call succeeds, and `reject` is only called if the `writeFile` call fails.

**Actual behavior:** The `reject` call is reported:

> Promise should not be resolved multiple times. Promise is potentially resolved

### Versions

- Node version: 16.17.0
- ESLint version: 8.25.0
- eslint-plugin-promise version: 6.1.0

### Additional Information

Repo to reproduce (but the rule is not enabled there yet):
https://github.com/OpenLightingProject/open-fixture-library/blob/6ee07f7a4a4be9907c1238531acd3ec0deeabdb3/plugins/qlcplus_4.12.2/exportTests/fixture-tool-validation.js#L81-L101

**Comment by ota-meshi** [comment]
Thank you for posting issue.

It seems a bug. We can see it in an [online demo] I personally made. I will look for a way to fix it.


[online demo]: https://ota-meshi.github.io/eslint-plugin-promise-playground/#eyJjb2RlIjoiZnVuY3Rpb24gZm4oKSB7XG5yZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICBodHRwcy5nZXQoc29tZVVybCwgcmVzcG9uc2UgPT4ge1xuICAgIHJlc3BvbnNlLm9uKGBlbmRgLCBhc3luYyAoKSA9PiB7XG4gICAgICB0cnkge1xuICAgICAgICBhd2FpdCB3cml0ZUZpbGUoc29tZUZpbGVQYXRoLCBzb21lQ29udGVudHMpO1xuICAgICAgICByZXNvbHZlKCk7XG4gICAgICB9XG4gICAgICBjYXRjaCAoZXJyb3IpIHtcbiAgICAgICAgcmVqZWN0KGVycm9yKTtcbiAgICAgIH1cbiAgICB9KTtcbiAgfSk7XG59KTtcbn0iLCJydWxlcyI6eyJwcm9taXNlL25vLW11bHRpcGxlLXJlc29sdmVkIjoyfX0=

**Comment by Romick2005** [comment]
Same here
![image](https://user-images.githubusercontent.com/17779660/195602326-9f35aba3-4478-453a-8a63-7f2d06e1841a.png)

**Comment by burtek** [comment]
@ota-meshi

```js
try {
    resolve(someThrowingFn())
} catch (error) {
    reject(error)
}
```

still errors for me. It doesn't error if I do this:

```js
try {
    const data = someThrowingFn()
    resolve(data)
} catch (error) {
    reject(error)
}
```

But those two are identical and should behave same way?

**Comment by sequencerr** [comment]
> @ota-meshi
> 
> ```js
> try {
>     resolve(someThrowingFn())
> } catch (error) {
>     reject(error)
> }
> ```
> 
> still errors for me. It doesn't error if I do this:
> 
> ```js
> try {
>     const data = someThrowingFn()
>     resolve(data)
> } catch (error) {
>     reject(error)
> }
> ```
> 
> But those two are identical and should behave same way?

yes, the same for me
```js
try {
    resolve({ data: await someThrowingFn() })
} catch (error) {
    reject(error)
}
```

**Comment by burtek** [comment]
@sequencerr opened as new issue
